app/models/service/aggregation.rb

```diff
@@ -42,26 +151,36 @@ def aggregate_direct_vm_memory_on_disk
   end
 
   def aggregate_all_vm_cpus
-    all_vms.inject(0) { |aggregate, vm| aggregate += vm.cpu_total_cores.to_i }
+    attributes["aggregate_all_vm_cpus"] || all_vms.inject(0) { |aggregate, vm| aggregate += vm.cpu_total_cores.to_i }
```

Comment:
> This runs into problems if the attribute is not defined.

@kbrock Not sure I follow the rational you have provided.  `attributes` is a hash, so if the hash key look up doesn't find a key, then it will continue down to the second part of the conditional and execute what was there originally.

That said, if you were arguing that this "stylistically is crap" or "not following conventions", then I would be fine on changing it for that reason alone.  Mostly did it this way to make the diff easier to grok as it only adds a check for a pre-calculated value first before moving on to the code that was there previously.

> Minor, but will `all_vms.inject(0) { |aggregate, vm| aggregate + vm.cpu_total_cores.to_i }` behave the same...

Yes, it should as far as I can tell, since the return value of `inject` should be the new value for the `aggregate`.  That said, I would argue if I am to do it, I should do it in a separate refactoring PR quickly first so I can touch the other aggregate methods in this PR that aren't part of this one.  Want me to do that?